tests/kolainst/destructive/state-overlays

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+## kola:
+##   tags: "needs-internet"
+
+set -euo pipefail
+
+. ${KOLA_EXT_DATA}/libtest.sh
+
+rm -rf /etc/yum.repos.d/*
+cat > /etc/yum.repos.d/vmcheck.repo << EOF
+[test-repo]
+name=test-repo
+baseurl=file:///${KOLA_EXT_DATA}/rpm-repos/0
+gpgcheck=0
+enabled=1
+EOF
+
+
+case "${AUTOPKGTEST_REBOOT_MARK:-}" in
+  "")
+    # switch over to local ref so upgrades are purely about package changes
+    booted_commit=$(rpm-ostree status --json | jq -r '.deployments[0].checksum')
+    ostree refs --create kolatest "${booted_commit}"
+    systemctl stop rpm-ostreed
+    unshare -m /bin/bash -c 'mount -o rw,remount /sysroot && sed -i -e "s/refspec=.*/refspec=kolatest/" /ostree/deploy/*/deploy/*.origin'
+
+    # XXX: until ostree v2024.1 hits FCOS
+    ostree_ver=$(rpm -q ostree  --qf '%{version}')
+    if [ "${ostree_ver}" != "2024.1" ] && \
+       [ "$(echo -e "${ostree_ver}\n2024.1" | sort -V | tail -n1)" = "2024.1" ]; then
+      rpm-ostree override replace https://bodhi.fedoraproject.org/updates/FEDORA-2024-6c7480dd2f
+    fi
+
+    # FCOS doesn't enable opt-usrlocal-overlays so use the hack instead
+    mkdir -p /etc/systemd/system/rpm-ostreed.service.d/
+    cat > /etc/systemd/system/rpm-ostreed.service.d/state-overlay.conf <<EOF
+[Service]
+Environment=RPMOSTREE_EXPERIMENTAL_FORCE_OPT_USRLOCAL_OVERLAY=1
```

Comment:
Ohhh I see it was for CI testing...ah.  WDYT about changing this test to use a container derivation instead?  Which could be done in-kola, although...

Or alternatively...this may argue for changing our CI here to have a separate flow which does separate treecompose runs and separate CI off each of them.

Reply:
Doing this from a container requires https://github.com/ostreedev/ostree-rs-ext/issues/573, no? Or do you mean fixing the directories and enabling the services in a container build but still installing the RPM client-side? That feels like the worst of both worlds. :)

Not too concerned about ergonomics for this. A service dropin isn't too bad a way to opt into experimental stuff, no? Though definitely once we have support in the container path, we should add a test for that.

Re. doing a separate treecompose run, yeah I think we do want that too (and has definitely come up before). Hmm, how about for now instead of adding another run to CI, we exercise it in the `fcos-e2e` tests since that one also does a full compose and kola tests? We can adapt the test to skip the experimental knob injection if it detects that it's already on in the base compose.